Author: Leigh Scott <leigh123linux@fedoraproject.org>
Date: Tue, 5 Oct 2021 15:25:50 +0100
Description: [PATCH] Fix menu-editor with python-3.10 (#10370)

Fix https://bugzilla.redhat.com/show_bug.cgi?id=2010737

  File "/usr/share/cinnamon/cinnamon-menu-editor/cme/util.py", line 22, in <module>
    from collections import Sequence
ImportError: cannot import name 'Sequence' from 'collections' (/usr/lib64/python3.10/collections/__init__.py)

Origin: https://github.com/linuxmint/cinnamon/commit/749a4952b7d5ad6dd258017a3d13b47351148953
---
 files/usr/share/cinnamon/cinnamon-menu-editor/cme/util.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/files/usr/share/cinnamon/cinnamon-menu-editor/cme/util.py b/files/usr/share/cinnamon/cinnamon-menu-editor/cme/util.py
index b5aedb4e5a..1343b9cbf9 100644
--- a/files/usr/share/cinnamon/cinnamon-menu-editor/cme/util.py
+++ b/files/usr/share/cinnamon/cinnamon-menu-editor/cme/util.py
@@ -19,7 +19,11 @@
 import os
 import xml.dom.minidom
 import uuid
-from collections import Sequence
+import sys
+if sys.version_info[:2] >= (3, 8):
+    from collections.abc import Sequence
+else:
+    from collections import Sequence
 from gi.repository import Gtk, GdkPixbuf, CMenu, GLib, Gdk
 
 DESKTOP_GROUP = GLib.KEY_FILE_DESKTOP_GROUP
